test/invariant/PROPERTIES.md

```diff
@@ -0,0 +1,137 @@
+# Properties
+
+## Unit
+
+| Tested                              | Property                                                                                                    |
+| -                                   | ----------------------------------------------------------------------------------------------------------- |
+| [yes](test/unit/BFactory.t.sol:113) | BFactory should always be able to deploy new pools                                                          |
+| [yes](test/unit/BFactory.t.sol:139) | BFactory's blab should always be modifiable by the current blabs                                            |
+| [yes](test/unit/BFactory.t.sol:198) | BFactory should always be able to transfer the BToken to the blab, if called by it                          |
+| [yes](test/unit/BPool.t.sol:1553)   | the amount received can never be less than min amount out                                                   |
+| [yes](test/unit/BPool.t.sol:1871)   | the amount spent can never be greater than max amount in                                                    |
+
+## Variable transition
+
+| Tested                           | Property                                                                                                    |
+| -                                | ----------------------------------------------------------------------------------------------------------- |
+| [yes](test/unit/BToken.t.sol:38) | BToken increaseApproval should increase the approval of the address by the amount                           |
```

Comment:
@drgorillamd is this kind of testing we're performing with foundry good enough? should we cover it with echidna as well?

Reply:
Nope, that's the idea: fuzzing one param gives a ok-answer (small fuzzing space + stateless), fuzzing campaign/echidna gives a better answer (bigger space + statefull), symb exec gives a math certainty (whole input space is covered, but stateless only)

Reply:
all in all, there is no perfect test, demonstrating safe and soundness is impossible, *but* we reduce the hole size in every cheese layer, then combine layers hoping no holes align

Reply:
![image](https://github.com/defi-wonderland/balancer-v1-amm/assets/83670532/3dacaf3a-643b-407d-982b-4d1baa7cf0e8)

Reply:
thanks for the explaination :heart: